app/src/App.tsx

```diff
@@ -109,6 +111,14 @@ export const App = () => {
     setIsModalOpen(true);
   }
 
+  const onClickCheckButton = (task: Task) => {
+    task.finishedAt = new Date().toISOString();
+    updateTaskMutation.mutate({
+      taskId: task.id, 
+      data: task
+    });
+  }
```

Comment:
```suggestion
  const onClickCheckButton = (task: Task) => {
    const finishedAt = new Date().toISOString();
    updateTaskMutation.mutate({
      taskId: task.id, 
      data: {
        ...task,
        finishedAt,
      }
    });
  }
```
コードを予測するのが難しくなるので、代入しないで↑な感じにすると良いと思います

Reply:
https://github.com/haruya3/kenshu-frontend-react/pull/3#discussion_r1198397552
コメントの通りにinvalidateQueriesかsetQueryDataを直すとうまく動くと思います！